backends/platform/ds/ds-graphics.cpp

```diff
@@ -406,7 +406,7 @@ void OSystem_DS::updateScreen() {
 	       SpriteColorFormat_Bmp, _cursorSprite, 0, false, !_cursorVisible, false, false, false);
 	oamUpdate(&oamMain);
 
-	if (_overlay.isVisible()) {
+	if (_overlayInGUI) {
```

Comment:
The `_overlay.isVisible()` check should be kept here, and the else should be replaced with a separate `_framebuffer.isVisible()` check.

Reply:
OK, after looking at the Background code I think I understood what you meant.

I kept the same order as, if I understood correctly, each background object is assigned a layer and updating one before the other makes no difference.

Thanks for removing this TODO! :)